packages/core/src/browser/shell/tab-bars.ts

```diff
@@ -66,3 +165,208 @@ export class TabBarRenderer implements TabBar.IRenderer<any> {
         }
     }
 }
+
+/**
+ * A specialized tab bar for side areas.
+ */
+export class SideTabBar extends TabBar<Widget> {
+
+    private static readonly DRAG_THRESHOLD = 5;
+
+    /**
+     * Emitted when a tab is added to the tab bar.
+     */
+    readonly tabAdded = new Signal<this, { title: Title<Widget> }>(this);
+    /**
+     * Side panels can be collapsed by clicking on the currently selected tab. This signal is
+     * emitted when the mouse is released on the selected tab without initiating a drag.
+     */
+    readonly collapseRequested = new Signal<this, Title<Widget>>(this);
+
+    private mouseData?: {
+        pressX: number,
+        pressY: number,
+        mouseDownTabIndex: number
+    };
+
+    constructor(options?: TabBar.IOptions<Widget>) {
+        super(options);
+
+        // Create the hidden content node (see `hiddenContentNode` for explanation)
+        const hiddenContent = document.createElement('ul');
+        hiddenContent.className = HIDDEN_CONTENT_CLASS;
+        this.node.appendChild(hiddenContent);
+    }
+
+    /**
+     * Tab bars of the left and right side panel are arranged vertically by rotating their labels.
+     * Rotation is realized with the CSS `transform` property, which disrupts the browser's ability
+     * to arrange the involved elements automatically. Therefore the elements are arranged explicitly
+     * by the TabBarRenderer using inline `height` and `top` styles. However, the size of labels
+     * must still be computed by the browser, so the rendering is performed in two steps: first the
+     * tab bar is rendered horizontally inside a _hidden content node_, then it is rendered again
+     * vertically inside the proper content node. After the first step, size information is gathered
+     * from all labels so it can be applied during the second step.
+     */
```

Comment:
For more information on the problems caused by the CSS `transform` attribute see https://drafts.csswg.org/css-transforms/#transform-rendering :

> Transformations do affect the visual rendering, but have no affect on the CSS layout other than affecting overflow.